src/transformers/trainer.py

```diff
@@ -880,8 +927,9 @@ def prediction_loop(
         # inside a DistributedDataParallel as we'll be under `no_grad` anyways.
 
         batch_size = dataloader.batch_size
+        num_examples = self.num_examples(dataloader) if is_dataset_sized else self.args.eval_steps * batch_size
```

Comment:
`self.args.eval_steps` is the number of training steps between two evaluation phases, so it can't be used here for that purpose. Since there is an error when `eval_dataset` is not sized, there should not be any test here.